test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -932,24 +923,10 @@ def run_cmd(self, hosts=None, cmd=None, sudo=False, stdin=None,
 
         ret = {'out': '', 'err': '', 'rc': 0}
 
-        user_hosts = self.get_user_hosts()
-        check_runas = False
-        if runas is not None:
-            if runas in user_hosts.keys():
-                for h in user_hosts.values():
-                    hostname_port = h.split('+')
-                    hostname = hostname_port[0]
-                    if len(hostname_port) > 1:
-                        port = hostname_port[1]
-                check_runas = True
-
         for hostname in hosts:
-            if not check_runas:
-                hostname_port = hostname.split('+')
-                hostname = hostname_port[0]
-                port = None
-                if len(hostname_port) > 1:
-                    port = hostname_port[1]
+            if (runas is not None) and (runas in PBS_ALL_USERS):
+                    hostname = runas.host
```

Comment:
I still don't understand this... Why we are overriding user given hostname? here `hosts` is list of hosts on which user wants to run command.
I understand the port part that ssh might be running on a given host on a different port than default (22) so while running the command we need to use given port but why hostname, by overriding hostname, we are forcing run_cmd to run all command for a given user to only one host.

Reply:
So I understand the requirement correctly then what we need it hostname vs ip+port mapping, correct? Like for hostname xyz IP is 1.2.3.4 and port is 8022 so whenever run_cmd try to run a command on host xyz it should IP 1.2.3.4 and port 8022 instead xyz... correct?

If my understanding is correct here then we are completely changing wrong place

Reply:
> we are forcing run_cmd to run all command for a given user to only one host.

That's what we want, all run_cmds for non-root users must be ran on their specific host (what I'll call a userhost) .

The userhosts have no hostname, they can only be accesed by an ip and a port. So if a user wants to run a command, it HAS TO run on their userhost.

Reply:
In that case I don't think we can make this change for all platform becase Linux doesn't work that way, right? so we have to make this change specific to platform who uses configuration which you said (user's commands must be run on userhost)

Reply:
If it's only for shasta that's fine, we can use the DshUtils.get_platform() function to determine if the machine is a shasta.

Reply:
@vstumpf I have already implemented this change. Please review.